be/src/util/thrift_rpc_helper.cpp

```diff
@@ -69,27 +103,20 @@ Status ThriftRpcHelper::rpc(const std::string& ip, const int32_t port,
         LOG(WARNING) << "Connect frontend failed, address=" << address << ", status=" << status.message();
         return status;
     }
-    try {
-        try {
-            callback(client);
-        } catch (apache::thrift::transport::TTransportException& e) {
-            status = client.reopen(timeout_ms);
-            if (!status.ok()) {
-                LOG(WARNING) << "client reopen failed. address=" << address << ", status=" << status.message();
-                return status;
-            }
-            callback(client);
-        }
-    } catch (apache::thrift::TException& e) {
-        std::stringstream ss;
-        ss << "call frontend service failed, address=" << address << ", reason=" << e.what();
-        LOG(WARNING) << ss.str();
-        SleepFor(MonoDelta::FromMilliseconds(config::thrift_client_retry_interval_ms * 2));
-        // just reopen to disable this connection
-        (void)client.reopen(timeout_ms);
-        return Status::ThriftRpcError(ss.str());
+
+    // Try 2 times.
+    for (int i = 0; i < 2; i++) {
+        status = rpc_impl(callback, client, address);
+        if (status.ok()) return status;
+
+        LOG(WARNING) << status << ", tried times: " << i + 1;
+
+        SleepFor(MonoDelta::FromMilliseconds(config::thrift_client_retry_interval_ms));
+        // reopen failure will disable this connection to prevent it from being used again.
+        RETURN_IF_ERROR_WITH_WARN(client.reopen(timeout_ms),
```

Comment:
Before the pr, if reopen success, will always return the error msg of rpc_impl.
After the pr, if reopen failed, return the status of reopen?